src/group.rs

```diff
@@ -163,3 +163,509 @@ impl Group for RistrettoPoint {
         constant_time_eq::constant_time_eq(&self.to_arr(), &other.to_arr())
     }
 }
+
+#[cfg(feature = "p256")]
+impl Group for p256_::ProjectivePoint {
+    type ElemLen = generic_array::typenum::U33;
+    type Scalar = p256_::Scalar;
+    type ScalarLen = U32;
+    type UniformBytesLen = generic_array::typenum::U96;
+
+    fn from_scalar_slice(
+        scalar_bits: &GenericArray<u8, Self::ScalarLen>,
+    ) -> Result<Self::Scalar, InternalPakeError> {
+        Ok(Self::Scalar::from_bytes_reduced(scalar_bits))
+    }
+
+    fn random_nonzero_scalar<R: RngCore + CryptoRng>(rng: &mut R) -> Self::Scalar {
+        use p256_::elliptic_curve::Field;
+
+        Self::Scalar::random(rng)
+    }
+
+    fn scalar_as_bytes(scalar: Self::Scalar) -> GenericArray<u8, Self::ScalarLen> {
+        scalar.into()
+    }
+
+    fn scalar_invert(scalar: &Self::Scalar) -> Self::Scalar {
+        scalar.invert().unwrap_or(Self::Scalar::zero())
+    }
+
+    fn from_element_slice(
+        element_bits: &GenericArray<u8, Self::ElemLen>,
+    ) -> Result<Self, InternalPakeError> {
+        use p256_::elliptic_curve::group::GroupEncoding;
+
+        Option::from(Self::from_bytes(element_bits)).ok_or(InternalPakeError::PointError)
+    }
+
+    fn to_arr(&self) -> GenericArray<u8, Self::ElemLen> {
+        use p256_::elliptic_curve::group::GroupEncoding;
+
+        self.to_bytes()
+    }
+
+    fn hash_to_curve(
+        uniform_bytes: &GenericArray<u8, Self::UniformBytesLen>,
+    ) -> Result<Self, InternalPakeError> {
+        use num_bigint::{BigInt, Sign};
+        use p256_::elliptic_curve::group::prime::PrimeCurveAffine;
+        use p256_::elliptic_curve::sec1::FromEncodedPoint;
+        use p256_::{AffinePoint, EncodedPoint};
+        use std::str::FromStr;
+
+        // https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-11#section-8.2
+        // `p: 2^256 - 2^224 + 2^192 + 2^96 - 1`
+        let p = BigInt::from_str(
+            "115792089210356248762697446949407573530086143415290314195533631308867097853951",
+        )
+        .unwrap();
+        // `A: -3`
+        let a = BigInt::from(-3);
+        // `B: 0x5ac635d8aa3a93e7b3ebbd55769886bc651d06b0cc53b0f63bce3c3e27d2604b`
+        let b = BigInt::parse_bytes(
+            b"5ac635d8aa3a93e7b3ebbd55769886bc651d06b0cc53b0f63bce3c3e27d2604b",
+            16,
+        )
+        .unwrap();
+        // `L: 48`
+        const L: usize = 48;
+        // `Z: -10`
+        let z = BigInt::from(-10);
```

Comment:
These could be const, which isn't supported in Rust at the moment. lazy-static or similar could be used, which is currently only a dev-dependency.

Reply:
Done.